Test/watchdog.cpp

```diff
@@ -11,55 +12,19 @@
 
 // NS2 headers
 
-/* - - - - - - Functions - - - - - - */
-
-/* - - - - - - main - - - - - - *
- * Usage:
- *  Main function entered on runtime
- * 
- * Inputs:
- *  none
- *  
- * Outputs:
- *  None
- */
- 
- // I think I'm writing this as Arduino code not c++? 
-int aRegCurr = 17;
-int dRegCurr = 18;
-int dRegPG = 20; 
+int dogPin = 2; 
 
 void loop()
 {
     while(true) // run main loop forever
     {
-		// Check voltage regulator currents
-        aCurr = analogRead(aRegCurr);
-		dCurr = analogRead(dRegCurr);
-		dPG = analogRead(dRegPG);
-		
-		
-		// Print statements for confirmation
-		// TODO: Gotta be a better way to format these statements?
-		Serial.print('Analog Current Value [A]: ');
-		Serial.print(aCurr);
-		Serial.print('\n');
-		
-		Serial.print('Digital Current Value [A]: ');
-		Serial.print(dCurr);
-		Serial.print('\n');
-		
-		Serial.print('Digital Power Good [V]: ');
-		Serial.print(dPG);
-		Serial.print('\n');
-		
-		// if the power good signal returns around 3.3V, power is good
-		// TODO: I have no clue how to write an if statement in arduino
-		if (dPG > 3.25) && (dPG < 3.35)
-			Serial.print('Good to go')
-		end
+		// Send a 3.3V, 50ns width square wave to trigger dog
+		squareWave = 0; // TOOD: Construct square wave
```

Comment:
Doesn't the dog need to be fed to keep the system on? Wouldn't the purpose of this pulse be to prevent the watchdog from triggering, not the opposite?

Reply:
That got totally flipped up in my mind, you are correct. Sorry. A 50ns width square wave must be sent to pin 2 every 1.2 seconds or the voltage input to the Teensy will be shorted. 

So we actually need to develop a watchdog module and incorporate it into all the other tests.

This watchdog test can become a "reset" test. This should be able to be accomplished by a long enough pause. I'd like to see the software reset and enter a loop again, unless our plan is to manually trigger a "turn on" function after reset.

Reply:
Yes that's how a watchdog works. If it doesn't get fed it triggers a hard reset without consulting the software. I believe we have it configured on the board to just cycle the power but I'm not sure about that. @JashanChopra ?